pallets/gear/src/lib.rs

```diff
@@ -805,11 +810,12 @@ pub mod pallet {
                 .unwrap_or(false)
         }
 
-        /// Returns true if id is a program and the program has active status.
+        /// Returns true if `program_id` is that of a in active status or the builtin actor.
         pub fn is_active(program_id: ProgramId) -> bool {
-            ProgramStorageOf::<T>::get_program(program_id)
-                .map(|program| program.is_active())
-                .unwrap_or_default()
+            T::BuiltinRouter::provide().lookup(&program_id).is_some()
```

Comment:
can we not to `provide()` it here every time?
checkout the usage and maybe it will be even unnecessary, or pass the ref to router